main.go

```diff
@@ -94,6 +94,15 @@ func main() {
 	switch cfg.Provider {
 	case "aws":
 		p, err = provider.NewAWSProvider(domainFilter, zoneIDFilter, zoneTypeFilter, cfg.DryRun)
+	case "aws-sd":
+		// Check that TXT registry is not used with AWS SD.
+		// Ownership information is maintained automatically in SD API using CreatorRequestId field.
+		if cfg.Registry != "noop" {
+			log.Infof("No need to use \"%s\" registry with AWS ServiceDiscovery. Switching to \"noop\".", cfg.Registry)
+			cfg.Registry = "noop"
+		}
+		namespaceTypeFilter := provider.NewNamespaceTypeFilter(cfg.AWSZoneType)
```

Comment:
I suggest we don't initialize it here, instead pass cfg.AWSZoneType in the NewAWSSDProvider and use it there to construct correct namespace type filter. Therefore, whole of NamespaceTypeFilter should be become private

Reply:
Good point 👍 I will move the whole namespace filtering to the AWS SD provider.